java/java.lsp.server/vscode/src/propertiesView/propertiesHtmlBuilder.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+import * as vscode from 'vscode';
+import { Properties, Property } from "./controlTypes";
+
+export function makeHtmlForProperties(name: string, nonce: string, scriptUri: vscode.Uri, properties: Properties): string {
+    return `<!DOCTYPE html>
+    <html lang="en">
+    
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <meta http-equiv="Content-Security-Policy" content="default-src 'none'; script-src 'nonce-${nonce}';">
+        <title>${name}</title>
+    </head>
+    
+    <body>
+        <h1>${name} Properties</h1>
+        <vscode-divider></vscode-divider>
+        <table>
+            ${makePropertiesTable(properties)}
+            <tr>
+                <td colspan="2">
+                </td>
+            </tr>
+            <tr>
+                <td colspan="2" align="right" style="text-align: right;">
+                    <vscode-button id="save" appearance="primary">Save</vscode-button>
+                    <vscode-button id="cancel" appearance="secondary">Cancel</vscode-button>
+                </td>
+            </tr>
+        </table>
+        <script type="module" nonce="${nonce}" src="${scriptUri}"></script>
+    </body>
+    
+    </html>`
+};
+
+function wrapToTable(name: string, content: string, separator: string = ":"): string {
+    return `<tr><td align="right"><b>${name}${separator}</b></td><td align="left">${content}</td></tr>`;
+}
+
+function makePropertiesTable(properties: Properties): string {
+    let html = "";
+    for (const prop of properties.props) {
+        html += makePropAccess(prop);
+    }
+    return html;
+}
+
+function makePropAccess(prop: Property): string {
+    let out: string;
+    switch (prop.propType) {
+        case 'java.lang.String':
```

Comment:
We have these String literals used in `PropTypes`, `PropTypeMap` and here ... still there is not an enum/const that would give a compile-type symbol to these literals ?